packages/ucp/docs/getting-started.md

```diff
@@ -0,0 +1,361 @@
+
+To initialise `UcpPatternEnforcement` as this engine, the following code is required:
+
+```ts
+import { PolicyExecutor, UcpPatternEnforcement, UcpPlugin, MemoryUCRulesStorage } from "@solidlab/ucp";
+import { EyeJsReasoner } from "koreografeye";
+
+// load plugin
+const plugins = { "http://example.org/dataUsage": new UcpPlugin() }
+// instantiate koreografeye policy executor
+const policyExecutor = new PolicyExecutor(plugins)
+// ucon storage
+const uconRulesStorage = new MemoryUCRulesStorage();
+// load N3 Rules
+const response = await fetch('https://raw.githubusercontent.com/woutslabbinck/ucp-enforcement/main/rules/data-crud-rules.n3'); // loading from the github repo
+const n3Rules: string[] = [await response.text()]
+// instantiate the enforcer using the policy executor,
+const ucpEvaluator = new UcpPatternEnforcement(uconRulesStorage, n3Rules, new EyeJsReasoner([
+        "--quiet",
+        "--nope",
+        "--pass"]), policyExecutor)
+```
+
+At this point, the engine is ready to be used. Which means that now you can use the `calculateAccessModes` function to request the **grants** for following *action request*: "Ruben wants to know the age of Wout".
+
+```ts
+const accessModes = await ucpEvaluator.calculateAccessModes({
+    subject: "https://pod.rubendedecker.be/profile/card#me",
+    action: ["http://www.w3.org/ns/auth/acl#Read"],
+    resource: "urn:wout:age",
+    owner: "https://pod.woutslabbinck.com/profile/card#me"
+});
+console.log(accessModes);
+```
+
+Unfortunately, this results into an empty list `[]`. Which means no **grants** are given and thus Ruben cannot know the age of Wout.
+
+The reason for this is very simple, there is **no** *Usage Control Policy* in the storage.
+
+This can however be resolved by simply adding such a policy to the **Usage Control Rule Storage**:
+
+```ts
+@prefix odrl: <http://www.w3.org/ns/odrl/2/> .
+@prefix : <http://example.org/usageControlRule> .
+
+:permission
+  a odrl:Permission ;
+  odrl:action odrl:read ;
+  odrl:target <urn:wout:age> ;
+  odrl:assignee <https://pod.rubendedecker.be/profile/card#me> ;
+  odrl:assigner <https://pod.woutslabbinck.com/profile/card#me> .
+```
+
+To add this rule to the storage, the following code can be used:
+
+```ts
+const ucr = `@prefix odrl: <http://www.w3.org/ns/odrl/2/> .
+@prefix : <http://example.org/usageControlRule> .
+
+:permission
+  a odrl:Permission ;
+  odrl:action odrl:read ;
+  odrl:target <urn:wout:age> ;
+  odrl:assignee <https://pod.rubendedecker.be/profile/card#me> ;
+  odrl:assigner <https://pod.woutslabbinck.com/profile/card#me> .
+    `
+const policyStore = await turtleStringToStore(ucr);
+await uconRulesStorage.addRule(policyStore);
+```
+
+From now on, when the access modes are calculated again, the following grants are received:
+
+```sh
+[ 'http://www.w3.org/ns/auth/acl#Read' ]
```

Comment:
Something that is not clear from these docs, is why you suddenly use ACL terminology, when you've been talking about ODRL.

Reply:
Good point, the reason for that is something we know: the goal for it was to be used in the CSS and the CSS understands these actions.

Do you have an idea how to incorporate that into the documentation?

Reply:
I'm not entirely sure, since it is not fully clear to me how close we should bind ourselves to ODRL.

My current opinion is that we should not need to do a mapping at all. The Resource Server uses certain access modes (e.g. from the ACP vocabulary, or the CSS's own mapping thereof), and mentions those when registering the resource at the Authorization Server, so it feels only natural to express the policies at the AS also in terms of those modes: the AS will somehow present those to the Resource Owner, who will make policies based on them, and so it's these modes that will end up in stored policies. For ODRL policies, that would probably mean using those modes as `odrl:action`.

In the short-term, for this PR, however, it might suffice to mention that the N3 rules make a mapping from ODRL action vocabulary to the one of the RS. Note, for example, that the N3 rules used in the current config on `main` already use mocked CSS-specific URNs.